Sources/SwiftSyntaxMacros/SwiftSyntaxMacros.docc/SwiftSyntaxMacros.md

```diff
@@ -1,37 +1,308 @@
+- [Building Syntax Nodes](#Building-Syntax-Nodes)
+- [Emitting Errors](#Emitting-Errors)
+
+
+## Macro Overview
+
+A new Swift Macro template can be created from Xcode or the command line. 
+
+- In Xcode, click File -> New -> Package… and select the *Swift Macro* template
+- On the command line, run `swift package init --type macro`
+
+Within the template, the `main.swift` file contains the following code.
+
+```swift
+let (result, code) = #stringify(a + b)
+```
+
+It uses an *expression macro*, indicated by the `#` character. At compile time, the compiler expands `#stringify(a + b)` to a tuple that contains the value of the macro’s argument as well as a string with the argument’s: `(a + b, "a + b")`. If `a = 17` and `b = 25` this evaluates to `(42, "a + b")` at runtime. In Xcode, the expanded code can be viewed by right-clicking on `stringify` and selecting *Expand Macro*.
+
+To expand the `stringify` macro, the compiler requires two ingredients: The macro declaration and the macro implementation.
+
+### Macro Declaration
+
+The macro’s declaration declares the macro’s signature, including its input and output type. The declaration of `stringify` looks as follows
+
+```swift
+@freestanding(expression)
+public macro stringify<T>(_ value: T) -> (T, String) = #externalMacro(module: "MyMacroMacros", type: "StringifyMacro")
+```
+
+`@freestanding(expression)` declares the macro with the freestanding expression macro role. This means that the macro can be used wherever an expression can be used and that its uses is indicated by the hash character, like `#stringify`. More macro roles are discussed below in [Macro Roles](#Macro-Roles).
+
+The next part of the macro declaration looks like a function signature. It takes a generic argument of type `T` and outputs a tuple containing the result, again of type `T`, as well as a string containing the source code of the macro’s argument. All arguments need to type check and the macro’s result type is inferred from the return type. If the arguments don’t type check or the result type doesn’t match its surrounding context, the compiler will emit an error without applying the macro expansion. This is different to, for example, C macros, which are evaluated at the pre-processor stage before type-checking.
+
+The last part of the macro links the macro declaration to the macro implementation. `#externalMacro` tells the compiler that to expand the macro, it needs to use the `StringifyMacro` type in the `MyMacroMacros` module.
+
+### Macro Implementation
+
+The `StringifyMacro` type is declared in a separate target, which is a compiler plug-in. The compiler will send the source code of the entire macro expression to that plug-in. The first thing that the macro plug-in does, is to parse the source code of the macro into a SwiftSyntax tree. This tree is a source-accurate, structural representation of the macro, and it will be the basis on which the macro operates. 
```

Comment:
Changed to 

> The compiler will send the macro expression to that plug-in. The plugin processes the macro expression as a SwiftSyntax tree, which is a source-accurate, structural representation of the macro, and it will be the basis on which the macro operates.